languages/python/django-oso/tests/conftest.py

```diff
@@ -22,3 +23,24 @@ def pytest_configure():
             "default": {"ENGINE": "django.db.backends.sqlite3", "NAME": ":memory:"}
         },
     )
+
+
+def negated_condition(variable):
+    if VERSION >= (3, 1):
+        return f"NOT {variable}"
+    else:
+        return f"{variable} = False"
+
+
+def parenthesize(variable):
+    if VERSION >= (3,):
+        return variable
+    else:
+        return f"({variable})"
+
+
+def is_true():
+    if VERSION >= (3, 1):
+        return ""
+    else:
+        return " = True"
```

Comment:
Oof. Not a huge fan. Would love to revisit as part of setting up tox or something similar.

Reply:
I don't think it changes the need for something like this if we're going to be checking against the raw sql

Reply:
Right, and we decided not to check against the Django filters because stringifying the top-level filter can't see through the nested Subquery/Exists objects...

Not sure there's a better strategy than this.